includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,319 @@
+			'links' => [
+				[
+					'type' => 'link-authentication',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-anon-sign-in',
+					],
+					'href' => $this->getHref( 'user-signin' ),
+					'param-name' => 'redirect',
+				],
+				[
+					'type' => 'link-authentication',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-anon-register',
+					],
+					'subtitle' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-anon-register-description',
+					],
+					'href' => $this->getHref( 'user-register' ),
+					"param-name" => "redirect"
+				],
+			],
+		];
+	}
+
+	private function getLoggedInUserData( $user ) {
+		global $wgEnableWallExt;
+
+		$userName = $user->getName();
+
+		return [
+			'header' => [
+				'type' => 'avatar',
+				'username' => [
+					'type' => 'text',
+					'value' => $userName
+				],
+				'url' => AvatarService::getAvatarUrl( $userName, 50 ),
+			],
+			'links' => [
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( $userName, NS_USER ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-view-profile'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $wgEnableWallExt
+						? $this->getPageUrl( $userName, NS_USER_WALL )
+						: $this->getPageUrl( $userName, NS_USER_TALK ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => $wgEnableWallExt
+							? 'global-navigation-user-message-wall'
+							: 'global-navigation-user-my-talk'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Preferences', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-my-preferences'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Contents', NS_HELP ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-help'
+					]
+				],
+				[
+					'type' => 'link-authentication',
+					'href' => $this->getPageUrl( 'UserLogout', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-sign-out'
+					],
+					'param-name' => 'returnto'
+				]
+			],
+		];
+	}
+
+	private function getNotifications( $user ) {
+		global $wgEnableWallExt;
+
+		$userName = $user->getName();
+
+		return [
+			'header' => [
+				'type' => 'line-image',
+				'image' => 'wsd-icons-bell',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-notifications-title'
+				]
+			],
+			'module' => [
+				'type' => 'notifications',
+				'url' => $wgEnableWallExt
+					? $this->getPageUrl( $userName, NS_USER_WALL )
+					: $this->getPageUrl( $userName, NS_USER_TALK )
+			]
+		];
+	}
+
+	private function getVerticalsSection() {
+		return [
+			'links' => [
+				[
+					'type' => 'link-branded',
+					'brand' => 'tv',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "brand" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)